src/main/kotlin/frc/robot/subsystems/gripper/Gripper.kt

```diff
@@ -0,0 +1,50 @@
+package frc.robot.subsystems.gripper
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase
+import edu.wpi.first.wpilibj.Timer
+import edu.wpi.first.wpilibj2.command.Command
+import org.littletonrobotics.junction.AutoLogOutput
+import org.littletonrobotics.junction.Logger
+
+
+class Gripper private constructor(private val io: GripperIO): SubsystemBase() {
+    private val timer = Timer()
+    @AutoLogOutput
+    private var rollerPowerSetPoint = 0.0
+
+    companion object {
+        @Volatile
+        private var instance: Gripper? = null
+
+        fun initialize(io: GripperIO) {
+            synchronized(this) {
+                if (instance == null) {
+                    instance = Gripper(io)
+                }
+            }
+        }
+
+        fun getInstance() : Gripper {
+            return instance ?: throw IllegalArgumentException(
+                "Gripper has not been initialized. Call initialize(io: GripperIO) first."
+            )
+        }
+    }
+
+    init {
+        timer.start()
+        timer.reset()
+    }
+
+    fun setRollerPower(power: Double): Command {
+        return run {
+            rollerPowerSetPoint = power
+            io.setRollerMotorPower(power)
+        }.withName("Set Roller Power")
+    }
+
+    override fun periodic() {
+        io.updateInputs()
+        if (timer.advanceIfElapsed(0.1)) Logger.processInputs(this::class.simpleName, io.inputs)
```

Comment:
Always wrap condition bodies in `{}`.

Reply:
65e573d0f6fe9cfcb7d53247c534546dfbc52805

Reply:
Do you really need to process the inputs every 0.02 seconds? It can cause problems.